app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -148,10 +179,27 @@ function FullAddressSearch({
       input(target.value);
     };
 
+  const inputChangeHandlerForZipCode =
+    <T extends HTMLElement & { value: string }>(input) =>
+    (event: React.ChangeEvent<T>) => {
+      const { target } = event;
+      const scrubbedZip = target.value.replace(/[^0-9]/g, '');
+      if (scrubbedZip.length > 5) {
+        const zipcodePieces: string[] = [];
+        zipcodePieces.push(scrubbedZip.substring(0, 5));
+        zipcodePieces.push(scrubbedZip.substring(5));
+        input(zipcodePieces.join('-'));
+      } else {
+        input(scrubbedZip);
+      }
+    };
```

Comment:
@NavaTim  Yesterday you and I talked about keydown for zipcode. Clicking the input somewhere other than the first position puts you at the first position when input is blank- else it adjust as I'd expect. I am wondering if we are solving for something that is not a problem. 

Also, I cannot find anywhere in idp that we use keydown. I wonder if that is something that you feel we strongly need to use rather than on change, maybe better to open an issue to document every form that needs updated and then open up issue for each form? I think the update as we go approach will leave inconsistencies in our code base so worth updating properly if there is a strong reason to do so.  Thoughts?

Reply:
> Clicking the input somewhere other than the first position puts you at the first position when input is blank- else it adjust as I'd expect.

The case I would recommend testing is attempting to insert a non-digit in the middle of an already-entered zip code because that's the case where we may alter the input value (so the cursor might jump). I'm not particularly concerned about the blank case.

> Also, I cannot find anywhere in idp that we use keydown.

I suspect a different approach could involve using [`setSelectionRange`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/setSelectionRange) to restore the user's cursor location.

> I wonder if that is something that you feel we strongly need to use rather than on change, maybe better to open an issue to document every form that would need updating and then open up issues for each form?

Most inputs also don't modify the input value as it's being typed, so I feel that the cause of the issue is also atypical. If it's difficult to prevent the cursor from jumping, then I would prefer that we avoid modifying the input value while the user is typing. The behavior is not a requirement for the story.